src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -3862,18 +3883,68 @@ public static void prohibitPartialHalfSlices(Design design) {
                 }
             }
 
+            // LUTs appear before their corresponding FFs in the iteration sequence so we
+            // can depend on any vacant LUT BELs being populated by the time we need to
+            // evaluate action for the FF BELs
             for (BEL bel : si.getSite().getBELs()) {
                 if (bel.getBELClass() == BELClass.BEL && si.getCell(bel) == null) {
                     Boolean isTop = isUltraScaleSliceTop(bel);
                     if (isTop != null) {
                         if ((isTop && topUsed) || (!isTop && bottomUsed)) {
-                            bels.add(si.getSiteName() + "/" + bel.getName());
+                            if (useDummyCells) {
+                                if (bel.isLUT()) {
+                                    // Being used as a physical static source
+                                    Net lutOutputNet = si
+                                            .getNetFromSiteWire(LUTTools.getLUTOutputPin(bel).getSiteWireName());
+                                    if (lutOutputNet != null) {
+                                        continue;
+                                    }
+
+                                    if (bel.getName().charAt(1) == '5') {
+                                        Net net = si.getNetFromSiteWire(bel.getName().charAt(0) + "6");
+                                        if (net != null) {
+                                            continue;
+                                        }
+                                    }
+
+                                    String name = "dummy_lut_" + dummyCount++;
+                                    Cell c = design.createAndPlaceCell(top, name, Unisim.LUT1,
+                                            si.getSiteName() + "/" + bel.getName());
+                                    c.getEDIFCellInst().addProperty("INIT", "1");
+                                    c.getEDIFCellInst().addProperty(EDIFTools.DONT_TOUCH, value);
+                                    c.removePinMapping("A6");
+                                    c.removePinMapping("A5");
```

Comment:
Is a LUT1 expected to have A6 and A5 pin mappings, or are you removing them both to cover both the `?6LUT` and `x5LUT` case?

Reply:
Just removing both to cover both cases.  I could check for what type of cell it is, but that might be a tad bit slower.

Reply:
Right. It's the creation placing and placing of a LUT that creates such pin mappings by default, right?

Reply:
Yes, this removes the default mappings.